File: algorithm/quickSort.c
#include<stdio.h>

void quickSort(int arr[],int m,int n){

    int temp = arr[m];
    int i=m,j=n;
    while(i<j){
        while(j>i && arr[j]>temp){
            --j;
        }
        if(j>i){
            arr[i] = arr[j];
        }
        while(i<j && arr[i]<temp){
            ++i;
        }
        if(i<j){
            arr[j] = arr[i];
        }
    }
    arr[i] = temp;
    quickSort(arr,m,i-1);
    quickSort(arr,i+1,n);
}

void main(){
    int arr[8] = {2,5,3,4,1,7,8,6};
    int i;
    quickSort(arr,0,7);

    for(i=0;i<8;++i){
        printf("%d ",arr[i]);
    }
    printf("\n");
}

File: algorithm/quickSort.js
/*
* iteration method
*/
function QuickSort(arr, left, right){
	var temp, i=left, j=right;

	if(left<right){
		temp=arr[left];

		while(i != j){
			while(j>i && arr[j]>temp) --j;
			if(i<j){
				arr[i]=arr[j];
				++i;
			}
			while(i<j && arr[i]<temp) ++i;
			if(i<j){
				arr[j]=arr[i];
				--j;
			}
		}
		arr[i]=temp;
		QuickSort(arr,left,i-1);
		QuickSort(arr,i+1,right);
	}
}

var input = [3,1,2,7,4,5,8,6];
var n = input.length;

console.log(input);
QuickSort(input, 0, n-1);
console.log(input);
input = null;